src/handlers/assign.rs

```diff
@@ -536,32 +539,61 @@ pub(super) async fn handle_command(
     Ok(())
 }
 
-#[derive(Debug)]
+#[derive(PartialEq, Debug)]
 enum FindReviewerError {
     /// User specified something like `r? foo/bar` where that team name could
     /// not be found.
     TeamNotFound(String),
-    /// No reviewer could be found. The field is the list of candidate names
-    /// that were used to seed the selection. One example where this happens
-    /// is if the given name was for a team where the PR author is the only
-    /// member.
-    NoReviewer(Vec<String>),
+    /// No reviewer could be found.
+    ///
+    /// This could happen if there is a cyclical group or other misconfiguration.
+    /// `initial` is the initial list of candidate names.
+    NoReviewer { initial: Vec<String> },
+    /// All potential candidates were excluded. `initial` is the list of
+    /// candidate names that were used to seed the selection. `filtered` is
+    /// the users who were prevented from being assigned. One example where
+    /// this happens is if the given name was for a team where the PR author
+    /// is the only member.
+    AllReviewersFiltered {
+        initial: Vec<String>,
+        filtered: Vec<String>,
+    },
 }
 
 impl std::error::Error for FindReviewerError {}
 
 impl fmt::Display for FindReviewerError {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> Result<(), fmt::Error> {
         match self {
-            FindReviewerError::TeamNotFound(team) => write!(f, "Team or group `{team}` not found.\n\
-                \n\
-                rust-lang team names can be found at https://github.com/rust-lang/team/tree/master/teams.\n\
-                Reviewer group names can be found in `triagebot.toml` in this repo."),
-            FindReviewerError::NoReviewer(names) => write!(
-                f,
-                "Could not determine reviewer from `{}`.",
-                names.join(",")
-            ),
+            FindReviewerError::TeamNotFound(team) => {
+                write!(
+                    f,
+                    "Team or group `{team}` not found.\n\
+                    \n\
+                    rust-lang team names can be found at https://github.com/rust-lang/team/tree/master/teams.\n\
+                    Reviewer group names can be found in `triagebot.toml` in this repo."
+                )
+            }
+            FindReviewerError::NoReviewer { initial } => {
+                write!(
+                    f,
+                    "No reviewers could be found from initial choice `{}`\n\
```

Comment:
I personally find the term "initial choice" confusing here. Is there a way to make it more clear what that means? Perhaps: "No reviewers could be found from initial request `{}`"

Reply:
Yea, I struggled with trying to write something that anyone would understand without making the error handling too complicated (this PR is already a bit larger than I wanted). Part of the issue is that there are two ways those "initial choices" come in, either by an explicit `r?` or via auto-detection from the diff and owners table. I didn't want to carry that information around just to make this error better.

Hopefully nobody ever sees this message. I think it should only happen if there is a misconfiguration.

I applied your suggestion, but if you have other ideas I'd be happy to add them.